components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -364,6 +389,76 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
         return OAuthUtil.buildConsumerAppDTO(app);
     }
 
+    private AuthenticatedUser getLoggedInUser(String tenantDomain) {
+
+        String tenantAwareLoggedInUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        AuthenticatedUser loggedInUser = null;
+        if (StringUtils.isNotEmpty(tenantAwareLoggedInUsername)) {
+            loggedInUser = buildAuthenticatedUser(tenantAwareLoggedInUsername, tenantDomain);
+        } else {
+            Optional<User> tenantAwareLoggedInUser = Optional.empty();
+            try {
+                tenantAwareLoggedInUser = OAuthUtil.getUser(tenantDomain, null);
+            } catch (IdentityApplicationManagementException e) {
+                LOG.error("Error while getting the user from the tenant domain: " + tenantDomain, e);
+            }
+            if (tenantAwareLoggedInUser.isPresent()) {
+                loggedInUser = new AuthenticatedUser(tenantAwareLoggedInUser.get());
+            }
+        }
+        return loggedInUser;
+    }
+
+    private static Map<String, Object> buildSPData(OAuthAppDO app) {
+
+        if (app == null) {
+            return new HashMap<>();
+        }
+        Gson gson = new Gson();
+        String oauthApp = maskSPData(app);
+        return gson.fromJson(oauthApp, new TypeToken<Map<String, Object>>() {
+        }.getType());
+    }
+
+    private static String maskSPData(OAuthAppDO oAuthAppDO) {
+
+        if (oAuthAppDO == null) {
+            return StringUtils.EMPTY;
+        }
+        try {
+            JSONObject oauthAppJSONObject =
+                    new JSONObject(new ObjectMapper().writeValueAsString(oAuthAppDO));
+            maskClientSecret(oauthAppJSONObject);
+            maskAppOwnerUsername(oauthAppJSONObject);
+            return oauthAppJSONObject.toString();
+        } catch (JsonProcessingException | IdentityException e) {
+            LOG.error("Error while converting service provider object to json.");
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private static void maskAppOwnerUsername(JSONObject oauthAppJSONObject) throws IdentityException {
+
+        JSONObject appOwner = oauthAppJSONObject.optJSONObject("appOwner");
+        if (!LoggerUtils.isLogMaskingEnable || appOwner == null) {
+            return;
+        }
+        String username = (String) appOwner.get("userName");
+        if (StringUtils.isNotBlank(username)) {
+            appOwner.put("userName", LoggerUtils.getMaskedContent(username));
+        }
+    }
+
+    private static void maskClientSecret(JSONObject oauthApp) {
+
+        if (oauthApp.get("oauthConsumerSecret") == null) {
+            return;
+        }
+        String secret = oauthApp.get("oauthConsumerSecret").toString();
+        String maskedSecret = secret.replaceAll(MASKING_REGEX, MASKING_CHARACTER);
```

Comment:
Any reason not to use `LoggerUtils.getMaskedContent` https://github.com/sahandilshan/carbon-identity-framework/blob/c265d493f395c0937e56a9e84c9625126341a6df/components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java#L195 here?
It has been used in `maskAppOwnerUsername` method

Reply:
fixed